src/Storages/StorageDistributed.cpp

```diff
@@ -976,9 +989,7 @@ void StorageDistributed::read(
 
     if (settings[Setting::allow_experimental_analyzer])
     {
-        StorageID remote_storage_id = StorageID::createEmpty();
-        if (!remote_table_function_ptr)
-            remote_storage_id = StorageID{remote_database, remote_table};
+        StorageID remote_storage_id = StorageID{remote_database, remote_table};
```

Comment:
This is a bit controversial change, because for table functions it is always `system.one`.